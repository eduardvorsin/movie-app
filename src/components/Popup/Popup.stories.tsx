import type { Meta, StoryObj } from '@storybook/react';
import Popup, { Props } from './Popup';
import { MouseEventHandler, useState } from 'react';
import Button from '../Button/Button';

const meta: Meta<typeof Popup> = {
	title: 'UI/Popup',
	tags: ['autodocs'],
	argTypes: {
		children: {
			description: 'The text content of the component',
		},
		placement: {
			description: 'Allows you to set the physical location along the vertical and horizontal axes relative to the anchor element.'
		},
		trigger: {
			description: 'The element that will show this component when interacting',
		},
		lockScroll: {
			description: 'Allows you to limit the ability to scroll the page when the popup is open',
		},
		testId: {
			description: 'Id for testing the component',
		},
		id: {
			description: 'A unique identifier for the component',
		},
		isOpen: {
			description: 'Reflects the open or closed state of the popup',
		},
		offsetX: {
			description: 'The displacement of the component along the X axis relative to the anchor element',
		},
		offsetY: {
			description: 'The displacement of the component along the Y axis relative to the anchor element',
		},
	},
	component: Popup,
	parameters: {
		docs: {
			description: {
				component: 'The Popup component refers to a popup window or element that appears on the screen on top of the main content. This component is usually used to display important information, additional settings, or dialog interaction with the user.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Popup>;

const PopupWithHooks = (props: Omit<Props, 'isOpen' | 'trigger'>) => {
	const [isOpen, setIsOpen] = useState<boolean>(false);

	const clickHandler: MouseEventHandler<HTMLButtonElement> = () => {
		setIsOpen((prevState) => !prevState);
	};

	return (
		<Popup
			{...props}
			className='w-[70px] h-[36px]'
			isOpen={isOpen}
			trigger={
				<Button
					onClick={clickHandler}
				>
					{isOpen ? 'Open' : 'Close'}
				</Button>
			}
		/>
	);
};

export const Default: Story = {
	args: {
		id: 'popup',
		placement: 'right-start',
		children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
	},
	parameters: {
		docs: {
			description: {
				story: 'The default state of the component',
			},
		},
	},
	render: (args) => (<PopupWithHooks {...args} />),
};

const placements: Exclude<Props['placement'], undefined>[] = ['left-start', 'left', 'left-end', 'top-start', 'top', 'top-end', 'right-start', 'right', 'right-end', 'bottom-start', 'bottom', 'bottom-end'];
export const Placement: Story = {
	args: {
		id: 'popup',
		placement: 'right-start',
	},
	parameters: {
		docs: {
			description: {
				story: 'The property sets the position of the hit relative to the anchor element. There are a total of 12 destination options available. There are 4 main directions: left, right, top, bottom. The remaining options are additional for placement relative to the main direction',
			},
		},
	},
	render: (args) => (
		<div className='flex items-center justify-center min-h-[250px] h-[100vh]'>
			<div className='max-w-[226px] grid grid-cols-[repeat(3,70px)] grid-rows-4 gap-2'>
				{placements.map((placement) => (
					<PopupWithHooks
						{...args}
						key={placement}
						placement={placement}
					>
						[{placement}] Lorem ipsum dolor sit amet consectetur adipisicing elit.
					</PopupWithHooks>
				))}
			</div>
		</div>
	),
};

export const LockScroll: Story = {
	args: {
		id: 'popup',
		placement: 'right-start',
		children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
		lockScroll: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'Restricts scrolling on the page when the popup is open',
			},
		},
	},
	render: (args) => (
		<div>
			<PopupWithHooks {...args} />

			<div className='mt-5'>
				{(new Array(15)).fill(null).map((_, index) => (
					<p
						key={index}
						className='mb-1 last:mb-0'
					>
						Lorem ipsum dolor sit amet consectetur, adipisicing elit. Doloremque optio dolorum reiciendis maiores, tempore soluta error maxime commodi est incidunt cum recusandae at eum culpa cupiditate id eos? Molestias, vitae?
						Voluptatum dolore illum quisquam, dicta facere et. Ipsam temporibus tempore aperiam, laudantium quae expedita nesciunt quas quos aliquid iste alias, quisquam debitis? Impedit sed inventore voluptatum ducimus quae eius asperiores?
					</p>
				))}
			</div>
		</div>),
};

export const offsetX: Story = {
	args: {
		id: 'popup',
		placement: 'right-start',
		children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
		offsetX: 100,
	},
	parameters: {
		docs: {
			description: {
				story: 'Allows you to set the X-axis offset by a certain number of pixels. The property is available when left-, right- is set for the placement property.',
			},
		},
	},
	render: (args) => (<PopupWithHooks {...args} />),
};

export const offsetY: Story = {
	args: {
		id: 'popup',
		placement: 'bottom-start',
		children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
		offsetY: 100,
	},
	parameters: {
		docs: {
			description: {
				story: 'Allows you to set the Y-axis offset by a certain number of pixels. The property is available when top-, bottom- is set for the placement property.',
			},
		},
	},
	render: (args) => (<PopupWithHooks {...args} />),
};