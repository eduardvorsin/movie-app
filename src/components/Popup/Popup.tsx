'use client';

import { GeneralProps } from '@/types/shared';
import { CSSProperties, ReactElement, ReactNode, useEffect, useRef, useState } from 'react';

export type Props = {
	children?: ReactNode,
	placement?: 'left-start' | 'left' | 'left-end' | 'top-start' | 'top' | 'top-end' | 'right-start' | 'right' | 'right-end' | 'bottom-start' | 'bottom' | 'bottom-end',
	trigger: ReactElement,
	lockScroll?: boolean,
	id: string,
	isOpen: boolean,
	offsetX?: number,
	offsetY?: number,
} & GeneralProps;

const placementVariants = {
	'left-start': '-left-[var(--offset-x)] -translate-x-full top-0 before:-right-5 before:rotate-90 before:top-[var(--triangle-y-offset)]',
	left: '-left-[var(--offset-x)] -translate-x-full top-1/2 -translate-y-1/2 before:-right-5 before:rotate-90 before:top-1/2 before:-translate-y-1/2',
	'left-end': '-left-[var(--offset-x)] -translate-x-full bottom-0 before:-right-5 before:rotate-90 before:bottom-[var(--triangle-y-offset)]',

	'right-start': '-right-[var(--offset-x)] translate-x-full before:-left-5 top-0 before:-rotate-90 before:top-[var(--triangle-y-offset)]',
	right: '-right-[var(--offset-x)] translate-x-full before:-left-5 top-1/2 -translate-y-1/2 before:-rotate-90 before:top-1/2 before:-translate-y-1/2',
	'right-end': '-right-[var(--offset-x)] translate-x-full before:-left-5 bottom-0 before:-rotate-90 before:bottom-[var(--triangle-y-offset)]',

	'top-start': 'bottom-[calc(100%+var(--offset-y))] left-0 before:rotate-180 before:-bottom-5 before:left-[var(--triangle-x-offset)] before:-translate-x-0.5',
	top: 'bottom-[calc(100%+var(--offset-y))] left-1/2 -translate-x-1/2 before:rotate-180 before:-bottom-5 before:left-1/2 before:-translate-x-1/2',
	'top-end': 'bottom-[calc(100%+var(--offset-y))] right-0 before:rotate-180 before:-bottom-5 before:right-[var(--triangle-x-offset)] before:translate-x-0.5',

	'bottom-start': 'top-[calc(100%+var(--offset-y))] left-0 before:left-[var(--triangle-x-offset)] before:-translate-x-0.5 before:-top-5',
	bottom: 'top-[calc(100%+var(--offset-y))] left-1/2 -translate-x-1/2 before:left-1/2 before:-translate-x-1/2 before:-top-5',
	'bottom-end': 'top-[calc(100%+var(--offset-y))] right-0 before:right-[var(--triangle-x-offset)] before:translate-x-0.5 before:-top-5',
} as const;

export default function Popup({
	className,
	id,
	testId,
	children,
	trigger,
	placement = 'bottom',
	isOpen,
	offsetX = 20,
	offsetY = 20,
	lockScroll: captureOverscroll,

}: Props) {
	const [containerSizes, setContainerSizes] = useState<{
		width: number,
		height: number,
	}>({
		width: 0,
		height: 0,
	});
	const containerRef = useRef<HTMLDivElement>(null);

	useEffect(() => {
		if (!containerRef.current) return;
		setContainerSizes({
			width: containerRef.current.clientWidth,
			height: containerRef.current.clientHeight,
		});
	}, []);

	useEffect(() => {
		const { documentElement } = document;
		const scrollbarWidth = window.innerWidth - documentElement.clientWidth;
		if (captureOverscroll && isOpen) {
			documentElement.style.overflow = 'hidden';
			documentElement.style.setProperty('--scrollbar-width', `${scrollbarWidth}px`);
		}

		return () => {
			documentElement.style.overflow = 'visible';
			documentElement.style.setProperty('--scrollbar-width', '0px');
		}
	}, [captureOverscroll, isOpen]);

	const wrapperClasses = [
		'relative',
		className
	].join(' ');

	const popupClasses = [
		'w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150',
		placementVariants[placement],
	].join(' ');

	return (
		<div
			id={id}
			ref={containerRef}
			className={wrapperClasses}
			data-testid={testId}
		>
			{trigger}

			{isOpen && (
				<div
					className={popupClasses}
					style={{
						'--triangle-x-offset': `${((containerSizes.width - 20) / 2).toFixed(2)}px`,
						'--triangle-y-offset': `${((containerSizes.height - 20) / 2).toFixed(2)}px`,
						'--offset-x': `${offsetX}px`,
						'--offset-y': `${offsetY}px`,
					} as CSSProperties}
				>
					{children}
				</div>
			)}
		</div>
	);
}