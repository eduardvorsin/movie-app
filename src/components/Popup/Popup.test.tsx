import { render, screen } from "@testing-library/react";
import Popup, { Props } from "./Popup";

const placements: NonNullable<Props['placement']>[] = ['left-start', 'left', 'left-end', 'top-start', 'top', 'top-end', 'right-start', 'right', 'right-end', 'bottom-start', 'bottom', 'bottom-end'];

describe('Popup tests', () => {
	it('is rendered correctly', () => {
		const mockFn = jest.fn();
		render(
			<Popup
				id='test-popup'
				isOpen={true}
				trigger={<button onClick={mockFn}>test trigger</button>}
				testId='test-popup'
			>
				test content
			</Popup>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-popup')).toBeInTheDocument();
	});

	it('is a basic snapshot', () => {
		const mockFn = jest.fn();
		render(
			<Popup
				id='test-popup'
				isOpen={true}
				trigger={<button onClick={mockFn}>test trigger</button>}
				testId='test-popup'
			>
				test content
			</Popup>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-popup')).toMatchSnapshot();
	});

	it.each(placements)('a snapshot with placement equal to %s', (placement) => {
		const mockFn = jest.fn();
		render(
			<Popup
				id='test-popup'
				isOpen={true}
				placement={placement}
				trigger={<button onClick={mockFn}>test trigger</button>}
				testId='test-popup'
			>
				test content
			</Popup>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-popup')).toMatchSnapshot();
	});

	it('is a snapshot with lockScroll set to true', () => {
		const mockFn = jest.fn();
		render(
			<Popup
				id='test-popup'
				isOpen={true}
				lockScroll
				trigger={<button onClick={mockFn}>test trigger</button>}
				testId='test-popup'
			>
				test content
			</Popup>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-popup')).toMatchSnapshot();
	});

	it('is a snapshot with the specified offsetX and offsetY', () => {
		const mockFn = jest.fn();
		render(
			<Popup
				id='test-popup'
				isOpen={true}
				offsetX={50}
				offsetY={100}
				trigger={<button onClick={mockFn}>test trigger</button>}
				testId='test-popup'
			>
				test content
			</Popup>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-popup')).toMatchSnapshot();
	});
});