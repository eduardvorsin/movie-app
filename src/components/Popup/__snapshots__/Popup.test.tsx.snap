// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup tests a snapshot with placement equal to bottom 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 top-[calc(100%+var(--offset-y))] left-1/2 -translate-x-1/2 before:left-1/2 before:-translate-x-1/2 before:-top-5"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to bottom-end 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 top-[calc(100%+var(--offset-y))] right-0 before:right-[var(--triangle-x-offset)] before:translate-x-0.5 before:-top-5"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to bottom-start 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 top-[calc(100%+var(--offset-y))] left-0 before:left-[var(--triangle-x-offset)] before:-translate-x-0.5 before:-top-5"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to left 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 -left-[var(--offset-x)] -translate-x-full top-1/2 -translate-y-1/2 before:-right-5 before:rotate-90 before:top-1/2 before:-translate-y-1/2"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to left-end 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 -left-[var(--offset-x)] -translate-x-full bottom-0 before:-right-5 before:rotate-90 before:bottom-[var(--triangle-y-offset)]"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to left-start 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 -left-[var(--offset-x)] -translate-x-full top-0 before:-right-5 before:rotate-90 before:top-[var(--triangle-y-offset)]"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to right 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 -right-[var(--offset-x)] translate-x-full before:-left-5 top-1/2 -translate-y-1/2 before:-rotate-90 before:top-1/2 before:-translate-y-1/2"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to right-end 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 -right-[var(--offset-x)] translate-x-full before:-left-5 bottom-0 before:-rotate-90 before:bottom-[var(--triangle-y-offset)]"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to right-start 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 -right-[var(--offset-x)] translate-x-full before:-left-5 top-0 before:-rotate-90 before:top-[var(--triangle-y-offset)]"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to top 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 bottom-[calc(100%+var(--offset-y))] left-1/2 -translate-x-1/2 before:rotate-180 before:-bottom-5 before:left-1/2 before:-translate-x-1/2"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to top-end 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 bottom-[calc(100%+var(--offset-y))] right-0 before:rotate-180 before:-bottom-5 before:right-[var(--triangle-x-offset)] before:translate-x-0.5"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests a snapshot with placement equal to top-start 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 bottom-[calc(100%+var(--offset-y))] left-0 before:rotate-180 before:-bottom-5 before:left-[var(--triangle-x-offset)] before:-translate-x-0.5"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests is a basic snapshot 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 top-[calc(100%+var(--offset-y))] left-1/2 -translate-x-1/2 before:left-1/2 before:-translate-x-1/2 before:-top-5"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests is a snapshot with lockScroll set to true 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 top-[calc(100%+var(--offset-y))] left-1/2 -translate-x-1/2 before:left-1/2 before:-translate-x-1/2 before:-top-5"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 20px; --offset-y: 20px;"
  >
    test content
  </div>
</div>
`;

exports[`Popup tests is a snapshot with the specified offsetX and offsetY 1`] = `
<div
  class="relative "
  data-testid="test-popup"
  id="test-popup"
>
  <button>
    test trigger
  </button>
  <div
    class="w-[200px] p-3 rounded-1 border-neutral-400 dark:border-dark-neutral-350 bg-neutral-100 dark:bg-dark-neutral-250 border-2 text-neutral-1000 dark:text-dark-neutral-1000 absolute z-300 before:pointer-events-none before:absolute before:w-5 before:h-5 before:[clip-path:polygon(50%_50%,0%_100%,100%_100%)] before:bg-neutral-400 dark:before:bg-dark-neutral-350 transition-colors duration-150 top-[calc(100%+var(--offset-y))] left-1/2 -translate-x-1/2 before:left-1/2 before:-translate-x-1/2 before:-top-5"
    style="--triangle-x-offset: -10.00px; --triangle-y-offset: -10.00px; --offset-x: 50px; --offset-y: 100px;"
  >
    test content
  </div>
</div>
`;
